code-challenges.test.js

```diff
@@ -12,33 +12,174 @@
 
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
+//Psuedocode
+//Input: Take in a string
+//Output: Return a string with a coded messege.
+//Process:  Create a function that takes in a given string and removes the selcted vowels.  Declare a object in the function that assign key (the vowels) value(numbers replacing the vowels). The string is split into a array of individual charecters using .split.  Use the .map to iterate over the array and checks to see if any of the letters matches the key in charMap, and if it does, replace it with the value from charMap. Finally, use .join to join the letters back into a string.  
+
+
 // a) Create a test with expect statements using the variables provided.
 
+// FAIL  ./code-challenges.test.js
+// words
+//   ✕ takes in a string and returns a string with a coded message, The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0 (1 ms)
+
+// ● words › takes in a string and returns a string with a coded message, The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0
+
+//   ReferenceError: words is not defined
+
+
+
 const secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+
+describe('words', () => {
+    it('takes in a string and returns a string with a coded message, The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0', () => {
+      expect(words('Lackadaisical')).toEqual('L4ck4d41s1c4l')
+      expect(words('Gobbledygook')).toEqual('G0bbl3dyg00k')
+      expect(words('Eccentric')).toEqual('3cc3ntr1c')
+    })
+  })
+
+const words = (string) => {
+    const charMap = {
+      'a': '4',
+      'e': '3',
+      'i': '1',
+      'o': '0'
+    }
+  
+    let words = string.split('').map((char) => {
+      return charMap[char.toLowerCase()] || char
+    }).join('')
+  
+    return words
+  }
+    const words2 = (string) => {
```

Comment:
No need to make 3 separate functions. You only need 1 functions, and then you call on that same function different times with different variables as arguments.